### DISCUSSION TITLE
Can I set the material opacity?

### DISCUSSION THREAD

**Discussion by Turtleted21** [google/model-viewer#2501] [Q&A]
Hello
Can I set the material opacity? I don't found a example,
 I look here https://github.com/google/model-viewer/blob/master/packages/model-viewer/src/features/scene-graph/material.ts
I tried `material.opacity = 0.5;` with no sucess

With three.js we can set the material opacity with a glb model.

three js editor test with DamagedHelmet.glb :
![modelviewer_opacity](https://user-images.githubusercontent.com/46889785/122915272-de4cec80-d35b-11eb-9eab-caad23c9870c.jpg)

Thanks

**Comment by elalish** [comment]
Yes, simply follow this [example](https://modelviewer.dev/examples/scenegraph/#changeColor) and set the alpha channel of your base color to your desired opacity.

**Reply by dmkrtz** [reply]
Following the example I was able to hide my texture by using `colorString = "0,0,0,-1";` and `.pbrMetallicRoughness.setBaseColorFactor(color);`.

By setting the BaseColorFactor I change the actual color of my texture, too. Is it possible to **only** modify the alpha channel?
Because by using "0,0,0,-1" for "hide" and "1,1,1,1" for "show" my result (toggling hide/show) is a white texture. 
Like, is there a different approach to go back to default BaseColorFactor settings?

Thanks in advance :)

**Reply by elalish** [reply]
I'm having some trouble following, but I think you want `[1, 1, 1, 0]` for hide and `[1, 1, 1, 1]` for show (this is the glTF default, which is white, but since it's a multiplier, it's also a pass-through for the texture color).

**Reply by dmkrtz** [reply]
Thank you for the reply.
Below is the original model + its material.

![sZJy9C5wW1](https://user-images.githubusercontent.com/89161020/130116172-7dab6dc4-a032-4338-9450-2b3aecfcf805.png)

By setting `[1, 1, 1, 0]` for hide, the result is a white color only. 

![AsadK9nrnK](https://user-images.githubusercontent.com/89161020/130116218-fa960045-b20d-4f62-bc83-d5e7e2f0e064.png)

By setting `[1, 1, 1, 1]`, nothing changes (looks same as above). 

If I use `[0, 0, 0, -1]` as my hide value, I achieve an invisible material, exactly what I want. 

![gG2hVRLaEB](https://user-images.githubusercontent.com/89161020/130116566-3baa6f33-44cc-4bb8-a736-14bbdddb1d3e.png)

But I need to find a way on how to reset the BaseColorFactor so the multiplier gets reset. A function like .resetBaseColorFactor() would be handy for that process.

**Reply by elalish** [reply]
Well, if you want to reset to your original baseColorFactor, just read the value first and save it to set later. `-1` is not a valid value, so who knows what's going  on  there. I bet  the reason alpha=0 is not working is probably because  you have the alphaMode set to Opaque. Try our new API: `material.alphaMode = 'BLEND'` or even `'MASK'`.

**Reply by dmkrtz** [reply]
Wow, I'm very disappointed by myself. Setting a BaseColorFactor on a textured material works flawless. To make something disappear I do have to use "0,0,0,-1", though. But "1,1,1,1" or setting the color before somewhere works as expected. 

Thanks, and sorry for the retardness, lol.

https://user-images.githubusercontent.com/89161020/130127294-fcf42dbf-a5fa-4e3c-8913-0a897737ffa8.mp4

**Reply by alien725** [reply]
@dmkrtz  hi , your work is great... I am trying to do some thing similar, but my scene , cannot switch properly when there are many models. Would you mind sharing part of your code that " how to specify the target model/ material"? thanks...

**Reply by elalish** [reply]
You may want `getMaterialByName()`, which you can find in our [Material API](https://modelviewer.dev/examples/scenegraph/#pickMaterialExample).

**Reply by alien725** [reply]
oh , i see, thank you!

**Comment by yosun** [comment]
hmm just checking if it is still possible to set alphaMode = 'blend' so that the a of rgba shows opacity?
